From 7f5788c3c456e904c782da6c5e02df2ddf40ada4 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 8 Mar 2023 15:42:09 +0530
Subject: [PATCH] Fix for HEVC rendering failing with usptream ffmpeg

With upstream FFmpeg setting of extended device id config filter
properties fails resulting in the HEVC rendering failure.

Extended device id are under ONEVPL_EXPERIMENTAL macro. As the
ONEVPL_EXPERIMENTAL is not enabled, vpl returns failure on
set config filter property request.

Define macro ONEVPL_EXPERIMENTAL to fix the issue.

Change-Id: I76ce4907bfa5658230ebb42846dc0651fba31aa4
Tracked-On: OAM-106567
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 android/mfx_defs.mk | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/android/mfx_defs.mk b/android/mfx_defs.mk
index 389dbfd1..2397ac3d 100644
--- a/android/mfx_defs.mk
+++ b/android/mfx_defs.mk
@@ -19,6 +19,7 @@ MEDIA_VERSION_EXTRA := ""
 MEDIA_VERSION_ALL := $(MEDIA_VERSION).pre$(MEDIA_VERSION_EXTRA)
 
 MFX_CFLAGS += -DMEDIA_VERSION_STR=\"\\\"${MEDIA_VERSION_ALL}\\\"\"
+MFX_CFLAGS += -DONEVPL_EXPERIMENTAL
 
 # Android version preference:
 ifneq ($(filter 12 12.% S ,$(PLATFORM_VERSION)),)
@@ -60,7 +61,8 @@ MFX_CFLAGS += \
   -fPIE -fPIC -pie \
   -O2 -D_FORTIFY_SOURCE=2 \
   -Wformat -Wformat-security \
-  -fexceptions -frtti -msse4.1
+  -fexceptions -frtti -msse4.1 \
+  -Wno-non-virtual-dtor
 
 # Enable feature with output decoded frames without latency regarding
 # SPS.VUI.max_num_reorder_frames
-- 
2.17.1

